src/core/dep-graph.ts

```diff
@@ -24,6 +24,10 @@ class DepGraphImpl implements types.DepGraphInternal {
   private _rootNodeId: string;
   private _rootPkgId: string;
 
+  private _pkgPathsToRootCache: Map<string, types.PkgInfo[][]> = new Map();
```

Comment:
I suggest we better not do the pkgPaths memoization in this PR.
This may be very costly memory-wise.

Reply:
b.t.w note I played with memorizing the node traversal here: https://github.com/snyk/dep-graph/tree/feat/more-efficient-paths-to-root - but I'm not sure it's really saving much CPU, and again there is that memory issue

Reply:
ok